### PR TITLE
Document using Nix to expose prebuilt Cabal packages

### PR DESCRIPTION
**Pull request by mboes**
This includes the following two new sections:

- Picking a compiler
- Building code with Hackage dependencies (using Nix)

### CODE REVIEW

File: docs/haskell-use-cases.rst [hunk 1]
```diff
@@ -3,6 +3,64 @@
...
+the `ghc_bindist`_ rule.
+
+The compiler can also be pulled from Nixpkgs_, a set of package
+definitions for the `Nix package manager`_. Pulling the compiler from
+Nixpkgs makes the build more hermetic, because the transitive closure
+of the compiler and all its dependencies is precisely defined in the
+``WORKSPACE`` file. Use `rules_nixpkgs`_ to do so (where ``X.Y.Z``
+stands for any recent release)::
+
+  http_archive(
```
Comment (regnat) [inline docs/haskell-use-cases.rst]: We could avoid the repetition of the release name with something like

```python
nixpkgs_release = "X.Y.Z"
http_archive(
    name = "io_tweag_rules_nixpkgs",
    strip_prefix = "rules_nixpkgs-{}".format(nixpkgs_release),
    urls = ["https://github.com/tweag/rules_nixpkgs/archive/v{}.tar.gz".format(nixpkgs_release)],
)

Comment (Profpatsch) [reply docs/haskell-use-cases.rst]: For the example it’s easier to see with the dummy, it acts like a “meta”-variable already, non?

File: docs/haskell-use-cases.rst [hunk 2]
```diff
@@ -3,6 +3,64 @@
...
+  )
+
+  nixpkgs_git_repository(
+      name = "nixpkgs",
+      revision = "18.09", # Any tag or commit hash
+  )
+
+  nixpkgs_package(
+      name = "ghc",
+      repositories = { "nixkgs": "@nixpkgs//:default.nix" }
```
Comment (Profpatsch) [inline docs/haskell-use-cases.rst]: typo `nixkgs`

File: docs/haskell-use-cases.rst [hunk 3]
```diff
@@ -38,5 +96,63 @@ This works for any ``haskell_binary`` or ``haskell_library`` target.
...
+detailed rationale.
+
+To use Nixpkgs in Bazel, we need `rules_nixpkgs`_. See `Picking
+a compiler`_ for how to import Nixpkgs rules into your workspace and
+how to use a compiler from Nixpkgs. To use Cabal packages from
+Nixpkgs, replace the compiler definition with the following::
+
+  nixpkgs_package(
+      name = "ghc",
+      repositories = { "nixkgs": "@nixpkgs//:default.nix" }
```
Comment (Profpatsch) [inline docs/haskell-use-cases.rst]: same typo

File: docs/haskell-use-cases.rst (hunk 1 above)
Comment (mboes) [reply docs/haskell-use-cases.rst]: @Profpatsch agreed.

**Comment by mboes** [comment]
@regnat but `haskell_import` _is_ covered. Just not as part of this PR, because it's already in `master` (see the tutorial).

**Comment by regnat** [comment]
> haskell_import is covered. Just not as part of this PR, because it's already in master

My bad, I didn't check long enough. Though I think it would be worth mentioning it again here because by just reading the use-case I have the impression that there isn't anything more to do and I can either directly refer to them in my `BUILD` files (but how?) or they will be implicitly added as a dependency to all my packages. That could be just one sentence such as

```rst
These packages may now be imported into bazel with the `haskell_import`_ macro.
```

**Comment by mboes** [comment]
@regnat PTAL

**Pull request merged**